mPower2/MotorControl/StepObjects/MCTInstructionStepObject.swift

```diff
@@ -0,0 +1,79 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+
+open class MCTInstructionStepObject : RSDUIStepObject, RSDNavigationSkipRule {
+    private enum CodingKeys: String, CodingKey {
+        case isFirstRunOnly
+    }
+    
+    public var isFirstRunOnly : Bool?
+    
+    /// Returns `true` if this step should be skipped, `false` otherwise.
+    /// Default: only returns `true` if the JSON file has isFirstRunOnly explicitly set
+    /// to true, and the conditions for a first run are not met. See
+    /// MCTOverviewStepViewController for more on what the conditions for a first run are.
+    public func shouldSkipStep(with result: RSDTaskResult?, conditionalRule: RSDConditionalRule?, isPeeking: Bool) -> Bool {
+        // if self.isFirstRunOnly == nil the JSON file probably left out this field so we
+        // assume that this step will always be shown
+        guard (self.isFirstRunOnly ?? false),
+              let isFirstRunResult = result?.findResult(with: MCTOverviewStepViewController.firstRunKey) as? RSDAnswerResult,
+              let isFirstRun = isFirstRunResult.value as? Bool
+            else {
+            return false
+        }
+        
+        // returns true if this isn't a first run and false otherwise.
+        return !isFirstRun
+    }
+    
+    /// Override decode to also decode whether this step is a first run only instruction.
+    /// If the "isFirstRunOnly" field is left out of the dictionary we are decoding, we default
+    /// to a value of `false`.
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.isFirstRunOnly = try container.decodeIfPresent(Bool.self, forKey: .isFirstRunOnly)
+        try super.init(from: decoder)
+    }
+    
+    /// Require method to initialize a MCTInstructionStepObject.
+    public required convenience init(identifier: String, type: RSDStepType?) {
+        self.init(identifier: identifier, type: type, isFirstRunOnly: false)
+    }
+    
+    /// Allows an MCTInstructionStepObject to be initialized and the isFirstRunOnly
+    /// property can be set. Intended for testing purposes.
+    internal init(identifier: String, type: RSDStepType?, isFirstRunOnly: Bool) {
```

Comment:
Since `isFirstRunOnly` is a nullable, readwrite property, you could also **not** have a custom initializer.

Instead, you can just override `open func decode(from decoder: Decoder, for deviceType: RSDDeviceType?) throws`.